ios/testmanagerd/testlistener.go

```diff
@@ -192,6 +227,27 @@ func (t *TestListener) Done() <-chan struct{} {
 	return t.executionFinished
 }
 
+func (t TestSuite) Close() error {
+	for _, testCase := range t.TestCases {
+		for _, attachment := range testCase.Attachments {
+			err := attachment.Close()
+			if err != nil {
+				return err
+			}
+		}
+	}
+
+	return nil
+}
+
+func (a TestAttachment) Close() error {
+	if _, err := os.Stat(a.Path); errors.Is(err, os.ErrNotExist) {
+		return nil
+	}
+
+	return os.Remove(a.Path)
+}
+
```

Comment:
And if we inject the path to the `TestListener` we don't need this, as the ownership moves. One can simply delete the directory once they are done
```suggestion
```

Reply:
Removed.